File: download.py
#!/usr/local/bin/python
# -*- coding: utf-8 -*-

import urllib
import urllib2
import os
import glob

from BeautifulSoup import BeautifulSoup as bs

MAX_PAGES = 50
HEADERS = {
    "User-Agent" : "Mozilla/5.0",
    "Accept" : "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
    "Accept-Language" : "en-US,en;q=0.8",
    "Cache-Control" : "no-cache",
    "Connection" : "keep-alive",
    "Pragma" : "no-cache"
}


def get_backgrounds():
    base = 'http://www.thefoxisblack.com/category/the-desktop-wallpaper-project/page/%s/'
    for x in range(0, MAX_PAGES):
        get_images_from_page(base % (x + 1))


def get_images_from_page(url):
    html = fetchurl(url)
    soup = bs(html)
    for link in soup.findAll('a'):
        try:
            href = link['href']
            if '-1680x1050' in href:
                fname = 'fox_backgrounds%s' % href[href.rfind('/'):]
                if not os.path.isfile(fname) and not os.path.isfile(fname + ".skip"):
                  print 'Downloading %s' % href
                  f = fetchurl(href)
                  with open(fname, "wb") as local_file:
                      local_file.write(f)
                else:
                  print 'Skipping %s' % href
        except (KeyError, urllib2.HTTPError):
            pass


def fetchurl(url):
    request = urllib2.Request(url, headers = HEADERS)
    response = urllib2.urlopen(request)
    content = response.read()
    return content


if __name__ == '__main__':
    get_backgrounds()

File: download.js
var cheerio = require('cheerio');
var request = require('request');
var http = require('http');
var async = require('async');
var fs = require('fs');
var path = require('path');

var maxPages = 50;
var existingFiles = fs.readdirSync("./fox_backgrounds").map(filenameWithoutSize);

var pageQueue = async.queue(function(url, callback) {
  request(url, function(err, resp, body){
    $ = cheerio.load(body);
    $(".paper").each(function() {
      var linkSet = [];
      $(this).find(".btn_download").each(function(j, link) {
        if ($(link).text().indexOf("×") > -1) {
          linkSet.push([$(link).text().split("×")[0], $(link).attr("href")]);
        }
      });
      linkSet.sort(function(a, b) { return parseInt(a[0]) - parseInt(b[0]); }).reverse();
      if (linkSet.length) {
        // linkSet[0] is the best link, download its href
        downloadQueue.push(linkSet[0][1]);
      }
    });
    callback();
  });
}, 1);

var downloadQueue = async.queue(function(url, callback) {
  // download and save file as long as we dont already have it.
  if (existingFiles.indexOf(filenameWithoutSize(path.basename(url))) === -1) {
    var file = fs.createWriteStream("./fox_backgrounds/" + path.basename(url));
    console.log("downloading", path.basename(url));
    file.on("close", callback);
    var req = http.get(url, function(res) {
      res.pipe(file);
    });
  } else {
    console.log("skipping", path.basename(url));
    callback();
  }
}, 2);

function getBackgrounds() {
  base = 'http://www.thefoxisblack.com/category/the-desktop-wallpaper-project/page/';
  for (var i = 1; i < maxPages; i++) {
    pageQueue.push(base + i + "/");
  }
}

function filenameWithoutSize(filename) {
  var match = filename.match(/^(.+)-(\d+x\d+)\.(.+)$/);
  if (match) { return match[1]; }
}

getBackgrounds();
